homework-submission/src/App.js

```diff
@@ -0,0 +1,136 @@
+import React, { Component } from "react";
+import "./App.css";
+import todosJSON from "./Sources/todosJSON.json";
+import List from "./Components/List";
+import NewTodoForm from "./Components/NewTodoForm";
+import uuid from "uuid/v4";
+
+const TODO_TEMPLATE = {
+  description: "",
+  deadline: null,
+  id: null,
+  done: false
+};
+
+export default class App extends Component {
+  state = {
+    todos: [...todosJSON],
+    updatedTodo: { ...TODO_TEMPLATE },
+    newTodo: {
+      ...TODO_TEMPLATE
+    }
+  };
+
+  handleToggleCheck = item => {
+    const todos = [...this.state.todos];
+    const todosCopy = todos.map(todo => {
+      if (todo.id !== item.id) return todo;
+      return { ...todo, done: !todo.done };
+    });
+
+    this.setState({ todos: todosCopy });
+  };
+
+  handleNewDescription = e => {
+    const newTodo = { ...this.state.newTodo };
+    newTodo.description = e.target.value;
+
+    this.setState({ newTodo });
+  };
+
+  handleNewDeadline = e => {
+    // is using a ref better here ? then combining this and handeleNewDescription in one go
```

Comment:
there are multiple techniques you can use to make this code re-usable but it would require an architectural change. I think it looks already very well :)